searx/webapp.py

```diff
@@ -490,6 +491,27 @@ def index():
     Supported outputs: html, json, csv, rss.
     """
 
+    def hash_client(request):
+        _ = hashlib.sha256()
+        _.update(request.headers.get('User-Agent').encode('utf-8'))
+        _.update(request.remote_addr.encode('utf-8'))
+        return _.digest()
+
+    if not 'searx_antibot' in session:
+        # install session cookie first / redirect same URL
+        # !! requires cookies to function properly !!
+        searx_antibot = hash_client(request)
+        logger.info('searx_antibot: await session cookie: %s ', searx_antibot)
+        session['searx_antibot'] = searx_antibot
+        return redirect(request.url)
```

Comment:
You need to redirect with a status code of 307 to keep the data sent by a POST method: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307
If you do a simple redirect with a status code of 302, the client/browser won't send the data again to the server and this will make the visitor kinda angry.

Reply:
Correct, there is a _jitter_ at the first time. 

> ... 307 guarantees that the method and the body will not be changed when the redirected request is made.

Was what I am looking for / thanks!